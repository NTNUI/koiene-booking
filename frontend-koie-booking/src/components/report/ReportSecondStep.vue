<template>
  <v-layout :class="$style.container" :dark="true">
    <h1 :class="$style.heading">{{ $t('report.step2') }}</h1>
    <v-form v-model="validForm" :class="$style.form">
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.supplyRegistration">{{ $t('report.gass_is_full') }}</h3>
        <v-layout class="px-4">
          <v-radio-group v-model="gasIsFull" mandatory required :color="$scssVars.globalColorBackgroundLight">
            <v-radio :value="true" :label="$t('report.gass_full')"></v-radio>
            <v-radio :value="false" :label="$t('report.gass_empty')"></v-radio>
          </v-radio-group>
        </v-layout>
      </v-layout>
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.form">{{ $t('report.firewood_supply') }}</h3>
        <v-layout class="px-4">
          <v-slider
            v-model="firewoodSupply"
            required
            :tick-labels="firewoodSupplyLabels"
            :max="4"
            step="1"
            ticks="always"
            tick-size="5"
          ></v-slider>
        </v-layout>
      </v-layout>
      <v-layout :class="$style.separator">
        <h3 class="py-4" :class="$style.form">{{ $t('report.chopped_up_wood_supply') }}</h3>
        <v-layout class="px-4">
          <v-slider
            v-model="choppedUpWoodSupply"
            required
            :tick-labels="choppedUpSupplyLabels"
            :max="4"
            step="1"
            ticks="always"
            tick-size="5"
          ></v-slider>
        </v-layout>
      </v-layout>
    </v-form>
  </v-layout>
</template>

<script lang="ts">
import Vue from 'vue';
import { ReportSecondStepData } from '../../types/report';

export default Vue.extend({
  name: 'ReportSecondStep',
  data(): ReportSecondStepData {
    return {
      validForm: true,
      firewoodSupplyLabels: [
        `${this.$t('report.firewood_supply_empty')}`,
        '',
        '',
        '',
        `${this.$t('report.firewood_supply_full')}`,
      ],
      choppedUpSupplyLabels: [
        `${this.$t('report.chopped_up_wood_supply_empty')}`,
        '',
        '',
        '',
        `${this.$t('report.chopped_up_wood_supply_full')}`,
      ],
    };
  },
  computed: {
    gasIsFull: {
      get() {
        return this.$store.state.report.reportData.gas_is_full;
      },
      set(value: boolean) {
        this.$store.commit('report/setGasIsFull', value);
        this.$store.commit('report/setEdited', true);
      },
    },
    firewoodSupply: {
      get() {
        return this.$store.state.report.reportData.firewood;
      },
      set(value: number) {
        this.$store.commit('report/setFirewoodSupply', value);
        this.$store.commit('report/setEdited', true);
      },
    },
    choppedUpWoodSupply: {
      get() {
        return this.$store.state.report.reportData.chopped_up_wood;
      },
      set(value: number) {
        this.$store.commit('report/setChoppedUpWoodSupply', value);
        this.$store.commit('report/setEdited', true);
      },
    },
  },
});
</script>

<style lang="scss" module>
.container {
  display: flex;
  flex-direction: column;
}
.heading {
  margin-top: 20px;
  align-self: center;
}
.separator {
  border-left: solid 4px;
  border-color: #2f3a50;
  display: flex;
  flex-direction: column;
  margin-top: 24px;
}
.separator > h3 {
  padding: 16px;
}
.form {
  padding-left: 16px;
  padding-right: 16px;
  @media only screen and (max-width: 600px) {
    padding-left: 8px;
    padding-right: 8px;
  }
}
.fieldWrapper {
  display: flex;
  width: 100%;
  align-items: center;
}
.numberField {
  padding-right: 10px;
}
</style>
