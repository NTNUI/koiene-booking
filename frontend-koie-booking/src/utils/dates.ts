import dayjs from 'dayjs';
require('dayjs/locale/nb');
dayjs.locale('nb');

export function getDateString(baseDate?: string, addDays?: number): string {
  let resultString: string;
  if (baseDate && addDays) {
    resultString = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else if (baseDate) {
    resultString = new Date(baseDate).toISOString().substr(0, 10);
  } else if (addDays) {
    resultString = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000).toISOString().substr(0, 10);
  } else {
    resultString = new Date().toISOString().substr(0, 10);
  }
  return resultString;
}

export function getDate(baseDate?: string, addDays?: number): Date {
  let resultDate: Date;
  if (baseDate && addDays) {
    resultDate = new Date(new Date(baseDate).getTime() + addDays * 24 * 60 * 60 * 1000);
  } else if (baseDate) {
    resultDate = new Date(baseDate);
  } else if (addDays) {
    resultDate = new Date(new Date().getTime() + addDays * 24 * 60 * 60 * 1000);
  } else {
    resultDate = new Date();
  }
  return resultDate;
}

export function addToDate(startDate: string, howMany: number, timeUnit: dayjs.OpUnitType): string {
  const date = dayjs(startDate + 'T00:00:00.000Z');
  return date.add(howMany, timeUnit).format('YYYY-MM-DD');
}

export function formatDate(dateISO: string, formatString: string): string {
  const date = dayjs(dateISO + 'T00:00:00.000Z');
  return date.format(formatString);
}
