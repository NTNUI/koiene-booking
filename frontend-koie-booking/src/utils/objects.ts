export const clone = (object: any) => Object.assign({}, object);

export const renameKey = (object: any, key: string, newKey: string): any => {
  const value = object[key];
  delete object[key];
  object[newKey] = value;
  return object;
};

export const sortObjectByKey = (unordered: any) => {
  let ordered: any;
  ordered = {};
  Object.keys(unordered)
    .sort()
    .forEach(function(key) {
      ordered[key] = unordered[key];
    });
  return ordered;
};
