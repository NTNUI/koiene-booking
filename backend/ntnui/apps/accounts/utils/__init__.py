from accounts.utils.mail_utils import send_gsuite_confirmation_mail
