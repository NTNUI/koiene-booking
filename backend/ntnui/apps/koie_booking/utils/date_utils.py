from django.utils.timezone import timedelta


def get_daterange(start_date, end_date):
    for n in range(int((end_date - start_date).days)):
        yield start_date + timedelta(n)
