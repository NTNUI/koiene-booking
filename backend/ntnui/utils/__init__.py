from ntnui.utils.pagination import PageSize7, PageSize20, PageSize100
