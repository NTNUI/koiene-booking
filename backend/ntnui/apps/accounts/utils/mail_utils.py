from django.core.mail import send_mail
from django.template.loader import render_to_string

from ntnui.enums import MembershipType


def new_membership(membership):
    """ Sends confirmation email to users and guests who signs up for events and sub-events. """

    # Mail header.
    sender = "noreply@mg.ntnui.no"
    receiver = [membership.member.email]
    subject = "Welcome"

    context = {"user": membership.member, "group": membership.group}

    # Mail body.
    plain_message = render_to_string("emails/new_membership.txt", context)
    html_message = render_to_string("emails/new_membership.html", context)

    send_mail(subject, plain_message, sender, receiver, html_message=html_message)


def send_gsuite_confirmation_mail(user, contact_email, password):
    # Mail header.
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = "Opprettelse av ny NTNUI-bruker"

    context = {
        "first_name": user.first_name,
        "email": user.email,
        "contact_email": contact_email,
        "password": password,
    }

    # Mail body.
    plain_message = render_to_string("emails/new_gsuite_account.txt", context)
    html_message = render_to_string("emails/new_gsuite_account.html", context)

    send_mail(subject, plain_message, sender, receiver, html_message=html_message)


def send_welcome_mail_to_leader(user, group):
    """ Send a mail to a new leader """
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny leder i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_leader_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_leader_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail_to_cashier(user, group):
    """Send a mail to the new cashier"""
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny kasserer i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_cashier_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_cashier_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail_to_board_member(user, group):
    """Send a mail to the new cashier"""
    sender = "noreply@mg.ntnui.no"
    receiver = [user.email]
    subject = f"Velkommen som ny tillitsvalgt i {group}"

    context = {
        "first_name": user.first_name,
        "email": user.get_contact_email(),
        "group": group.name,
    }

    plain_message = render_to_string(
        template_name="emails/new_board_membership.txt", context=context
    )
    html_message = render_to_string(
        template_name="emails/new_board_membership.html", context=context
    )

    send_mail(
        subject=subject,
        message=plain_message,
        from_email=sender,
        recipient_list=receiver,
        html_message=html_message,
    )


def send_welcome_mail(membership):
    """ Delegate welcome mail to different board members """
    send_welcome_mail_to_board_member(user=membership.member, group=membership.group)
    if membership.type == MembershipType.leader:
        send_welcome_mail_to_leader(user=membership.member, group=membership.group)

    if membership.type == MembershipType.cashier:
        send_welcome_mail_to_cashier(user=membership.member, group=membership.group)
