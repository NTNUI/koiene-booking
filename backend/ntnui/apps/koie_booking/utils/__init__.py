from koie_booking.utils.mail_utils import send_confirmation_mail, send_koie_information_mail
from koie_booking.utils.date_utils import get_daterange
