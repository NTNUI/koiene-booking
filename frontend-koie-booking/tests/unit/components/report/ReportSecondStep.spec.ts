import Vue from 'vue';
import Vuetify from 'vuetify';

Vue.use(Vuetify);

// Utilities
import { Wrapper, ThisTypedShallowMountOptions } from '@vue/test-utils';
import { createWrapper } from '../../utils';

// Components or views
import ReportSecondStep from '@/components/report/ReportSecondStep.vue';

describe('Component ReportSecondStep.vue', () => {
  let wrapper: Wrapper<any>;
  let wrapperOptions: ThisTypedShallowMountOptions<any>;

  beforeEach(() => {
    wrapperOptions = {};
    wrapper = createWrapper(ReportSecondStep, wrapperOptions);
  });

  it('Matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
