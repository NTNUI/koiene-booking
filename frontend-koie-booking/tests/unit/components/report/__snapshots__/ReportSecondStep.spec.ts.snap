// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component ReportSecondStep.vue Matches snapshot 1`] = `
<div class="layout dark container">
  <h1 class="heading">Supplies</h1>
  <form novalidate="novalidate" class="v-form form">
    <div class="layout separator">
      <h3 class="py-4">Was the gas full?</h3>
      <div class="layout px-4">
        <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-input--selection-controls v-input--radio-group v-input--radio-group--column">
          <div class="v-input__control">
            <div class="v-input__slot" style="height: auto;">
              <div role="radiogroup" aria-labelledby="input-9" class="v-input--radio-group__input">
                <div class="v-radio theme--light v-item--active">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-marked theme--light primary--text"></i><input aria-checked="true" id="input-10" role="radio" type="radio" name="radio-9" value="true">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-10" class="v-label theme--light" style="left: 0px; position: relative;">Yes, it was full</label>
                </div>
                <div class="v-radio theme--light">
                  <div class="v-input--selection-controls__input"><i aria-hidden="true" class="v-icon notranslate mdi mdi-radiobox-blank theme--light"></i><input aria-checked="false" id="input-12" role="radio" type="radio" name="radio-9" data-test="btnGasEmpty" value="false">
                    <div class="v-input--selection-controls__ripple primary--text"></div>
                  </div><label for="input-12" class="v-label theme--light" style="left: 0px; position: relative;">No, it was empty</label>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
          </div>
        </div>
      </div>
    </div>
    <div class="layout separator">
      <h3 class="py-4 form">How much firewood was there at the cabin?</h3>
      <div class="layout px-4">
        <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-input__slider">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-slider v-slider--horizontal theme--light"><input value="2" id="input-16" readonly="readonly" tabindex="-1" required="required">
                <div class="v-slider__track-container">
                  <div class="v-slider__track-background primary lighten-3" style="right: 0px;"></div>
                  <div class="v-slider__track-fill primary" style="left: 0px; width: 50%;"></div>
                </div>
                <div class="v-slider__ticks-container v-slider__ticks-container--always-show"><span class="v-slider__tick v-slider__tick--filled" style="width: 5px; height: 5px;"><div class="v-slider__tick-label">No logs</div></span><span class="v-slider__tick v-slider__tick--filled" style="width: 5px; height: 5px;"></span><span class="v-slider__tick" style="width: 5px; height: 5px;"></span><span class="v-slider__tick" style="width: 5px; height: 5px;"></span><span class="v-slider__tick" style="width: 5px; height: 5px;"><div class="v-slider__tick-label">A lot of logs</div></span></div>
                <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="4" aria-valuenow="2" aria-readonly="false" aria-orientation="horizontal" required="required" class="v-slider__thumb-container primary--text" style="left: 50%;">
                  <div class="v-slider__thumb primary"></div>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
          </div>
        </div>
      </div>
    </div>
    <div class="layout separator">
      <h3 class="py-4 form">Did you find a lot of chopped up wood at the cabin?</h3>
      <div class="layout px-4">
        <div class="v-input v-input--is-label-active v-input--is-dirty theme--light v-input__slider">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-slider v-slider--horizontal theme--light"><input value="2" id="input-19" readonly="readonly" tabindex="-1" required="required">
                <div class="v-slider__track-container">
                  <div class="v-slider__track-background primary lighten-3" style="right: 0px;"></div>
                  <div class="v-slider__track-fill primary" style="left: 0px; width: 50%;"></div>
                </div>
                <div class="v-slider__ticks-container v-slider__ticks-container--always-show"><span class="v-slider__tick v-slider__tick--filled" style="width: 5px; height: 5px;"><div class="v-slider__tick-label">No chopped up wood</div></span><span class="v-slider__tick v-slider__tick--filled" style="width: 5px; height: 5px;"></span><span class="v-slider__tick" style="width: 5px; height: 5px;"></span><span class="v-slider__tick" style="width: 5px; height: 5px;"></span><span class="v-slider__tick" style="width: 5px; height: 5px;"><div class="v-slider__tick-label">A lot of chopped up wood</div></span></div>
                <div role="slider" tabindex="0" aria-valuemin="0" aria-valuemax="4" aria-valuenow="2" aria-readonly="false" aria-orientation="horizontal" required="required" class="v-slider__thumb-container primary--text" style="left: 50%;">
                  <div class="v-slider__thumb primary"></div>
                </div>
              </div>
            </div>
            <div class="v-messages theme--light"><span name="message-transition" tag="div" class="v-messages__wrapper"></span></div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
